compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/StreamingCodeDesugar.java

```diff
@@ -784,14 +1111,121 @@ private BInvokableSymbol getNextProcessFunctionSymbol(BVarSymbol nextProcessInvo
         return recordKeyValueList;
     }
 
-    private BLangLambdaFunction createLambdaFunction(DiagnosticPos pos, BLangVariable lambdaFunctionVariable,
+    private BLangInvocation generateAggregatorInvocation(BVarSymbol typeCastedVariableSymbol,
+                                                         BVarSymbol streamEventSymbol, BVarSymbol aggregatorArraySymbol,
+                                                         long aggregatorIndex, BLangInvocation selectFuncInvocation) {
+        // aggregatorArr[0]
+        BLangIndexBasedAccess indexBasedAccess = createIndexBasedAggregatorExpr(aggregatorArraySymbol,
+                aggregatorIndex, selectFuncInvocation.pos);
+
+        // aggregatorArr[0].process(..)
+        BLangInvocation aggregatorInvocation = ASTBuilderUtil.createInvocationExpr(selectFuncInvocation.pos,
+                getNextProcessFunctionSymbol(indexBasedAccess.type.tsymbol), Collections.emptyList(), symResolver);
+        aggregatorInvocation.expr = indexBasedAccess;
+
+        // arguments of aggregatorArr[0].process(..). e.g. (t.age, e.eventType)
+        List<BVarSymbol> params = ((BInvokableSymbol) aggregatorInvocation.symbol).params;
+        aggregatorInvocation.requiredArgs = generateAggregatorInvocationArgs(typeCastedVariableSymbol,
+                streamEventSymbol, selectFuncInvocation, params);
+
+        return aggregatorInvocation;
+    }
+
+    private List<BLangExpression> generateAggregatorInvocationArgs(BVarSymbol typeCastedVariableSymbol,
+                                                                   BVarSymbol streamEventSymbol,
+                                                                   BLangInvocation funcInvocation,
+                                                                   List<BVarSymbol> params) {
+        // generates the fields which will be aggregated e.g. t.age
+        List<BLangExpression> args = generateAggregatorInputFieldsArgs(typeCastedVariableSymbol, funcInvocation,
+                params);
+        // generate the EventType for the aggregation o.eventType
+        BLangFieldBasedAccess streamEventFieldAccess = generateStreamEventTypeForAggregatorArg(streamEventSymbol,
+                funcInvocation.pos, params);
+        // (t.age, o.eventType)
+        args.add(streamEventFieldAccess);
+        return args;
+    }
+
+    private BLangFieldBasedAccess generateStreamEventTypeForAggregatorArg(BVarSymbol streamEventSymbol,
+                                                                          DiagnosticPos pos,
+                                                                          List<BVarSymbol> params) {
+        BLangFieldBasedAccess streamEventFieldAccess = createFieldBasedEventTypeExpr(streamEventSymbol, pos);
+        // always the 2nd parameter is the EventType, so the 2nd parameter's type should match.
+        streamEventFieldAccess.type = params.get(1).type;
```

Comment:
Lets move this `1` to a constant. Will be easy to maintain later if we needed to change anything.

Reply:
+1, fixed.